test/y2partitioner/dialogs/lvm_lv_info_test.rb

```diff
@@ -0,0 +1,179 @@
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../test_helper"
+
+require "cwm/rspec"
+require "y2storage"
+require "y2partitioner/dialogs"
+require "y2partitioner/sequences/controllers"
+
+describe Y2Partitioner::Dialogs::LvmLvInfo do
+  before do
+    devicegraph_stub("lvm-two-vgs.yml")
+  end
+
+  subject { described_class.new(controller) }
+
+  let(:controller) { Y2Partitioner::Sequences::Controllers::LvmLv.new(vg) }
+
+  let(:vg) { Y2Storage::LvmVg.find_by_vg_name(current_graph, "vg0") }
+
+  let(:current_graph) { Y2Partitioner::DeviceGraphs.instance.current }
+
+  include_examples "CWM::Dialog"
+
+  describe Y2Partitioner::Dialogs::LvmLvInfo::NameWidget do
+    subject(:widget) { described_class.new(controller) }
+
+    before do
+      allow(widget).to receive(:value).and_return(value)
+    end
+
+    let(:value) { nil }
+
+    include_examples "CWM::AbstractWidget"
+
+    describe "#validate" do
+
+      context "when no lv name is entered" do
+        let(:value) { nil }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+
+        it "return false" do
+          expect(widget.validate).to be(false)
+        end
+      end
+
+      context "when the entered lv name is not valid" do
+        let(:value) { "$!" }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+
+        it "return false" do
+          expect(widget.validate).to be(false)
+        end
+      end
+
+      context "when the entered lv name is already used" do
+        let(:value) { "lv1" }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+
+        it "return false" do
+          expect(widget.validate).to be(false)
+        end
+      end
+
+      context "when the entered lv name is valid" do
+        let(:value) { "lv3" }
+
+        it "does not show an error popup" do
+          expect(Yast::Popup).to_not receive(:Error)
+          widget.validate
+        end
+
+        it "return true" do
+          expect(widget.validate).to be(true)
+        end
+      end
+    end
+
+    describe "#store" do
+      let(:value) { "lv3" }
+
+      it "sets #lv_name in the controller" do
+        expect(controller.lv_name).to be_nil
```

Comment:
You can use the `expect { }.to change` form ,as IMHO is more expressive:

```ruby
expect { widget.store }.to change { controller.lv_name }.from(nil).to(value)
```

There are more places where you could use the same approach, but I am not going to write a comment for every one (maybe you are not interesting in applying this change).

Reply:
Nice, I did not know that, but at least in this case, I prefer the "classic" way. Yes, maybe I am too much familiar with this way.